### PR TITLE
Replace ModuleNotFoundError with the more general ImportError

### LINKED ISSUE #2436: ImportError vs ModuleNotFoundError
In the current source codes, sometimes we use `ImportError` and sometimes `ModuleNotFoundError`, e.g.,
https://github.com/GenericMappingTools/pygmt/blob/5fd85ddff29eba92c32e79a7efa08651b4c50f19/pygmt/__init__.py#L130

https://github.com/GenericMappingTools/pygmt/blob/5fd85ddff29eba92c32e79a7efa08651b4c50f19/pygmt/figure.py#L11

`ModuleNotFoundError` is a subclass of `ImportError` (https://docs.python.org/3/library/exceptions.html#ModuleNotFoundError). As per https://www.scivision.dev/python-importerror-vs-modulenotfound/, maybe we should always use `ImportError`.

ref: https://stackoverflow.com/questions/62854761/python3-8-whats-the-difference-between-importerror-and-modulenotfounderror

### PR DESCRIPTION
**Pull request by seisman**
**Description of proposed changes**

ref: https://stackoverflow.com/questions/62854761/

Fixes #2436.

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.
- [ ] Use underscores (not hyphens) in names of Python files and directories.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

**Review by michaelgrund — APPROVED** [review]

File: pygmt/datasets/tile_map.py [hunk 1]
```diff
@@ -109,10 +109,10 @@ def load_tile_map(region, zoom="auto", source=None, lonlat=True, wait=0, max_ret
     """
     # pylint: disable=too-many-locals
     if contextily is None:
-        raise ModuleNotFoundError(
+        raise ImportError(
             "Package `contextily` is required to be installed to use this function. "
             "Please use `pip install contextily` or "
-            "`conda install -c conda-forge contextily` "
+            "`mamba install -c conda-forge contextily` "
```
Comment (seisman) [inline pygmt/datasets/tile_map.py:115]: Will be changed in https://github.com/GenericMappingTools/pygmt/pull/2394, so revert the change here:
```suggestion
            "`conda install -c conda-forge contextily` "
```

**Comment by seisman** [comment]
Ping @weiji14 Are you OK with the changes?

**Review by weiji14 — APPROVED** [review]
